src/router/handlers/fedimint/ln/await_invoice.rs

```diff
@@ -23,54 +24,37 @@ pub struct AwaitInvoiceRequest {
     pub federation_id: Option<FederationId>,
 }
 
-async fn _await_invoice(
-    client: ClientArc,
-    req: AwaitInvoiceRequest,
-) -> Result<InfoResponse, AppError> {
-    let lightning_module = &client.get_first_module::<LightningClientModule>();
-    let mut updates = lightning_module
+pub async fn handle_ws(
+    state: AppState,
+    params: Value,
+    ws_sender: Sender,
+) -> Result<(), AppError> {
+    // Deserialize requested parameters
+    let req: AwaitInvoiceRequest = serde_json::from_value(params)
+        .map_err(|e| AppError::new(StatusCode::BAD_REQUEST, anyhow!("Invalid request: {}", e)))?;
+
+    let client = state.get_client(req.federation_id).await.map_err(|_| AppError::from(StatusCode::INTERNAL_SERVER_ERROR))?;
+
+    let mut updates = client
+        .get_first_module::<LightningClientModule>()
         .subscribe_ln_receive(req.operation_id)
-        .await?
+        .await.map_err(|_| AppError::from(StatusCode::INTERNAL_SERVER_ERROR))?
         .into_stream();
-    while let Some(update) = updates.next().await {
-        info!("Update: {update:?}");
-        match update {
-            LnReceiveState::Claimed => {
-                return Ok(get_note_summary(&client).await?);
-            }
-            LnReceiveState::Canceled { reason } => {
-                return Err(AppError::new(
-                    StatusCode::INTERNAL_SERVER_ERROR,
-                    anyhow!(reason),
-                ))
-            }
-            _ => {}
-        }
 
-        info!("Update: {update:?}");
+    while let Some(update) = updates.next().await {
+        let message = serde_json::to_string(&update).map_err(|_| AppError::from(StatusCode::INTERNAL_SERVER_ERROR))?;
+        ws_sender.send(Message::Text(message)).await.map_err(|_| AppError::from(StatusCode::INTERNAL_SERVER_ERROR))?;
```

Comment:
this is looking like the right direction but the request needs to be correlated to the response by the id, don't think that's being added in here. Also this is just putting text it should probably be json structured or use enums for the responses so they can be deterministically consumed by the client